### DISCUSSION TITLE
Alternative mechanism for detemining if DDS SHM is available between two clients

### DISCUSSION THREAD

**Discussion by traverseda** [eProsima/Fast-DDS#5348] [Ideas]
As per https://fast-dds.docs.eprosima.com/en/latest/docker/shm_docker.html 

>The way Fast DDS utilizes to find out whether a remote [DomainParticipant](https://fast-dds.docs.eprosima.com/en/latest/fastdds/dds_layer/domain/domainParticipant/domainParticipant.html#dds-layer-domainparticipant) is running on the same host as the local one is through a hashing of the network interfaces, which entails that two Docker container which are not sharing the same network stack will be detected as two different hosts, and therefore Fast DDS will defer to communicating through the [UDP Transport](https://fast-dds.docs.eprosima.com/en/latest/fastdds/transport/udp/udp.html#transport-udp-udp) instead.

Requiring containers run in network-mode host is a problem for many deployment types. I'd suggest instead reading from an environment variable, something like `FAST_DDS_SHM_NAMESPACE`  perhaps. This would make it a lot easier to manage security context for systems that use FastDDS with docker.

**Comment by JesusPoderoso** [comment]
Hi @traverseda, thanks for your contribution.
This feature is not part of our current roadmap but we will definitely take it into account for future enhancements to the library, thanks!

If you are interested in speeding this development up for future releases, do not hesitate to contact our commercial support support@eprosima.com.

**Reply by traverseda** [reply]
We'll keep that in mind, thanks
